static/js/util/api_actions.js

```diff
@@ -46,13 +47,30 @@ export const removeComment = R.curry(
     )
 )
 
-export const toggleFollowPost = R.curry((dispatch: Dispatch<*>, post: Post) =>
-  dispatch(actions.posts.patch(post.id, { subscribed: !post.subscribed }))
+export const toggleFollowPost = R.curry(
+  async (dispatch: Dispatch<*>, post: Post) => {
+    const newSubscribedValue = !post.subscribed
+    await dispatch(
+      actions.posts.patch(post.id, { subscribed: newSubscribedValue })
+    )
+    const message = newSubscribedValue
+      ? `Followed post ${post.title}. You will receive email for every comment.`
+      : `Unfollowed post ${
+        post.title
+      }. You will no longer receive email about this post.`
+    return dispatch(setSnackbarMessage({ message: message }))
+  }
 )
 
 export const toggleFollowComment = R.curry(
-  (dispatch: Dispatch<*>, comment: CommentInTree) =>
-    dispatch(
-      actions.comments.patch(comment.id, { subscribed: !comment.subscribed })
+  async (dispatch: Dispatch<*>, comment: CommentInTree) => {
+    const newSubscribedValue = !comment.subscribed
+    await dispatch(
+      actions.comments.patch(comment.id, { subscribed: newSubscribedValue })
     )
+    const message = newSubscribedValue
+      ? "Followed comment. You will receive email for every reply to this comment."
+      : "Unfollowed comment. You will no longer receive email about replies to this comment."
+    return dispatch(setSnackbarMessage({ message: message }))
```

Comment:
same comment about object, also I'm curious why you're returning the call to `dispatch(setSnackbarMessage(...))` here?

Reply:
No other reason than to keep consistency with the functions that I modified. They both returned the results of a `dispatch()` call before I modified them. If that's not important I'm happy to change it

Reply:
well, `setSnackbarMessage` is a synchronous action, so I think `dispatch(setSnackbarMessage)` will just evaluate to `undefined`, and since this is an `async` it always returned a promise anyway. I was just wondering if there was something I was missing that you were doing with the return value. I think you could either change it or leave it, it won't make too much difference

Reply:
actually I just checked at it returns the action! not `undefined` as I thought it did.